Objects/longobject.c

```diff
@@ -3051,6 +3094,47 @@ PyLong_AsDouble(PyObject *v)
 
 /* Methods */
 
+void
+_PyLong_ExactDealloc(PyObject *obj)
+{
+    assert(PyLong_CheckExact(obj));
+    PyLongObject *op = (PyLongObject *)obj;
+#if PyLong_MAXFREELIST > 0
+    if (!IS_MEDIUM_VALUE(op)) {
+        PyObject_Del(op);
```

Comment:
See summary; where we used to call `PyObject_Free` unconditionally, the conditional call changed to `PyObject_Del`, which is almost certainly incorrect. Same goes for all other uses of `PyObject_Del` here.